sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -201,7 +201,7 @@ def __init__(
             network_tracing=kwargs.get("logging_enable"),
             **kwargs
         )
-
+        self._auth_uri = f"sb://{self._address.hostname}{self._address.path}"
```

Comment:
this doesnt need the consumer group for producer?

Reply:
Producer + MgmtClient used for management operations don't need consumer group. It's only used for ReceiveClient ops. will remove!

Reply:
There's an operation in the ConsumerProducerMixin (def _open) that is shared by the async Producer and Consumer. We call [_create_auth in there](https://github.com/Azure/azure-sdk-for-python/blob/72919a99ab62cbdb0647493ea43241d8f3df84c7/sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py#L566), but the auth URI that should be passed in differs between Producer and Consumer. So we need a separate _auth_uri defined explicitly on producer vs. consumer.